external_deps/build.sh

```diff
@@ -759,7 +759,6 @@ setup_windows-amd64-mingw() {
 # Set up environment for 64-bit amd64 macOS
 setup_macos-amd64-default() {
 	HOST=x86_64-apple-darwin11
-	CROSS=
```

Comment:
Won't removing CROSS for some platforms cause errors because we have `set -u`? Looks like Lua uses CROSS unconditionally.

Reply:
Right, this one is entirely removed later. I now squashed that commit `it with `entirely remove CROSS variable`, I also extracted some CROSS-related editions from `factorize common platform setup` and squashed them in `entirely remove CROSS variable` as well. So now `entirely remove CROSS variable` does all the CROSS removal job and `factorize common platform setup` is simpler.